### PR TITLE
OneOf validator

### PR DESCRIPTION
**Pull request by philtay**
If these changes are ok I'll add the unit tests. Ref. issue #135

### CODE REVIEW

File: marshmallow/validate.py [hunk 1]
```diff
@@ -271,6 +274,48 @@ def __call__(self, value):
...
+        A sequence of valid values.
+    :param iterable labels:
+        Optional sequence of labels to pair with the choices.
+    :param str error:
+        Error message to raise in case of a validation error.
+    """
+    def __init__(self, choices, labels=None, error=None):
+        self.choices = choices
+        self.labels = labels if labels is not None else []
+        self.error = error or 'Not a valid choice.'
```
Comment (sloria) [inline marshmallow/validate.py]: Perhaps we can have a more descriptive error message that can be templated (like in `Range` and `Length`). Something like: `'{val} is not one of {choices}'`

**Comment by sloria** [comment]
Out of curiosity, can you describe how you plan to use the `pairs` method? How will you access it and for what purpose?

File: marshmallow/validate.py [hunk 2]
```diff
@@ -271,6 +274,48 @@ def __call__(self, value):
         return value
 
 
+class OneOf(object):
+    """Validator which succeeds if ``value`` is a member of ``choices``.
+
+    :param iterable choices:
```
Comment (sloria) [inline marshmallow/validate.py:280]: For consistency with the rest of the codebase, don't add an extra newline after `param_name:`.

**Comment by sloria** [comment]
Looks good so far. :+1: Will merge this after the tests are added.

**Comment by philtay** [comment]
> Perhaps we can have a more descriptive error message that can be templated (like in Range and Length). Something like: '{val} is not one of {choices}'

I thought about this and decided to go for a basic message. Two reasons:
- It's not very useful to show {val} because the end user already see what is selected in the UI (e.g. a select field). Furthermore, more often than not, {val} is an opaque value to the end user (e.g. you select United Kingdom and the error message would be "GB is not one of ...").
- Listing all the available {choices} can be a security vulnerability in some circumstances. Not something to do by default.

> Out of curiosity, can you describe how you plan to use the pairs method? How will you access it and for what purpose?

The purpose is to generate a select form field. Something like that:

``` html
<select>
  <option value="{value}">{label}<option>
  ...
  ...
</select>
```

The options can be generated using the `pairs` method. It's useful if you want to keep together valid values and their respective labels. If you can't specify the labels somewhere in Python you'll end up to hardcode them in the UI. To use the method you can instantiate the validator and pass it both to a schema field and to your HTML template. You can also generate the list of valid values and labels, convert to JSON and throw it at your preferred client side framework (e.g. Angular or Ember) that will take care of displaying a nice select for the happiness of your users! ;)

> For consistency with the rest of the codebase, don't add an extra newline after param_name:

Ok, but all the other validators' docs have the extra newline.

Please let me know how to move forward.

**Comment by sloria** [comment]
It would still be nice to have the option to interpolate `val` and `choices` if the user wanted to do that (even if the default message didn't use those values).

Thanks for the clarification re: `pairs`.

Yes, I realize now the validators follow a different convention from the rest of the codebase. I can make those consistent in a separate commit after this is merged.

**Comment by philtay** [comment]
Done. Let me know if it's ok and I'll add the tests. Tnx

**Comment by sloria** [comment]
Looks good. Once tests are added, this should be good to merge.

**Comment by philtay** [comment]
Done.

**Comment by sloria** [comment]
Thanks!

**Pull request merged**